packages/components/src/components/domainproperties/dataset/DatasetColumnMappingPanel.tsx

```diff
@@ -0,0 +1,163 @@
+            return this.compareNames('sequencenum', inferredFieldName) && inferredField.rangeURI === 'xsd:double';
+        } else if (targetColumnName == timepointType) {
+            return this.compareNames('date', inferredFieldName) && inferredField.rangeURI === 'xsd:datetime';
+        }
+    }
+
+    findClosestColumn(targetColumn: string): string {
+        const inferredColumns = this.props.model.domain.fields;
+
+        const matchedField = inferredColumns.find(field => {
+            let matchingNames = false;
+
+            if (field.name && field.rangeURI) {
+                matchingNames = this.areColumnNamesAndTypesEquivalent(targetColumn, field);
+            }
+            return matchingNames;
+        });
+
+        return matchedField ? matchedField.name : undefined;
+    }
+
+    onSelectChange = (name, formValue, selected): void => {
+        const value = selected ? selected.name : undefined;
+
+        this.setState(
+            () => ({ [name]: value }),
+            () => {
+                this.props.onColumnMappingChange(
+                    this.state.closestParticipantIdField,
+                    this.state.closestTimepointField
+                );
+            }
+        );
+    };
+
+    render() {
+        const { model } = this.props;
+        const { closestParticipantIdField, closestTimepointField } = this.state;
+
+        const domain = model.domain;
+
+        return (
+            <>
+                <SectionHeading title="Column mapping" />
+                <div className="margin-top">
+                    Columns already existing in the domain can be mapped with columns from your file. Choose a column to
+                    match your ParticipantID and Timepoints.
+                </div>
+                <Row className="margin-top">
+                    <Col xs={4}>
+                        <DomainFieldLabel label="ParticipantID" />
+                    </Col>
+                    <Col xs={5}>
+                        <SelectInput
+                            onChange={this.onSelectChange}
+                            value={closestParticipantIdField}
+                            options={domain.fields.toArray()}
+                            inputClass=""
+                            containerClass=""
+                            labelClass=""
+                            formsy={false}
+                            multiple={false}
+                            required={false}
+                            name="closestParticipantIdField"
+                            labelKey="name"
+                            valueKey="name"
+                            clearable={true}
+                        />
+                    </Col>
+                    <Col xs={3} />
+                </Row>
+                <Row className="margin-top">
+                    <Col xs={4}>
+                        <DomainFieldLabel label="Timepoint" />
```

Comment:
Similar to above, we'll want to have this label based on the study modulecontext timepoint type (i.e. either Visit or Timepoint)